### DISCUSSION TITLE
failing to find npm dependencies

### DISCUSSION THREAD

**Discussion by nagarajui7** [renovatebot/renovate#10462] [Request Help]
Hi Team,

Iam trying to check for npm dependencies, but getting the below mentioned error 

```
Failed to look up dependency fast-deep-equal (repository=repo-name, packageFile=packages/Authoring/package.json, dependency=fast-deep-equal)
```
Below are the different ways i tried
```
hostRules: [
                {
                        hostType: 'npm',
                        matchHost: 'https://myartifactory.com/artifactory/api/npm/npmjs-dev/',
                        token: process.env.NPM_TOKEN,
                        authType: "Basic"                  
                },],
npmrc: `registry=https://myartifactory.com/artifactory/api/npm/npmjs-dev/\nalways-auth=true\nemail=abc@gmail.com`

hostRules: [
                {
                        hostType: 'npm',
                        matchHost: 'https://myartifactory.com/artifactory/api/npm/npmjs-dev/',
                        token: process.env.NPM_TOKEN,             
                },],
npmrc: `registry=https://myartifactory.com/artifactory/api/npm/npmjs-dev/`

```

Renovate setup: Running as a docker container. version is 25.42.1

**Comment by rarkins** [comment]
Are you sure that Renovate knows to look up your private registry in the first place? i.e. is it looking up registry.npmjs.org or myartifactory.com.

Second question is: are you sure the token you provided works? e.g. have you tested it with `curl`? If so then you should know whether Bearer or Basic is right, and can narrow it down.

My advice is:
- Commit a `.npmrc` file to your repo which contains the `registry=` information and anything else required
- Add `hostRules` for the credentials part

**Reply by nagarajui7** [reply]
Are you sure that Renovate knows to look up your private registry in the first place? i.e. is it looking up registry.npmjs.org or myartifactory.com.

```
I have followed the document for setting npm as private registry. I could see the matchHost URL in debuglogs
````
Second question is: are you sure the token you provided works? e.g. have you tested it with curl? If so then you should know whether Bearer or Basic is right, and can narrow it down.
```
I have used the api token using curl command and it worked.
```
As suggested, I have added the `registry=` in `.npmrc` file

@rarkins, As part of hostRules, when we say `token` - is it just the API token or combination of api_token and username ?

**Reply by nagarajui7** [reply]
also, can u help me with the correct config for npm that can be added under hostRules ?

**Reply by rarkins** [reply]
Please provide the exact `curl` command you ran to fetch an example dependency and then we can tell you the configuration you need. Please keep any sanitization to a minimum - ideally just host name and the token, and keep everything else.

**Reply by nagarajui7** [reply]
sorry for the delay @rarkins 
Here is the curl command i have used:
`curl -u uname:<api-token> -X GET https://my-artifactory/artifactory/api/npm/npmjs-dev/%40pega/cosmos-react-work/-/cosmos-react-work-1.0.0.tgz -O`

**Reply by viceice** [reply]
`config.js`
```js
hostRules: [
                {
                        hostType: 'npm',
                        matchHost: 'https://myartifactory.com/artifactory/api/npm/npmjs-dev/',
                        username: 'uname',
                        password: process.env.ARTIFACTORY_APITOKEN,
                },
],
```

`.npmrc` (in target repo)
```ini
registry=https://myartifactory.com/artifactory/api/npm/npmjs-dev/
always-auth=true
email=abc@gmail.com
```

⚠️  Do not use `NPM_TOKEN` as env var.

**Reply by rarkins** [reply]
@viceice should that be `user` or `uname`? I don't know if `uname` is what they actually use or it's a sanitization.

**Reply by viceice** [reply]
Updated my sample to use env var to make more clear it needs to be adapted.

Ok, @nagarajui7  changed his comment too 😏

**Reply by viceice** [reply]
I'm not sure if `email` is really required. 🤔

**Reply by nagarajui7** [reply]
Thanks for the response @viceice @rarkins
